### PR TITLE
Run3-hcx352A Modify the sensitive detector xmls of HCAL to allow old style HF simhits

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify the sensitive detector xmls of HCAL to allow old style HF simhits

#### PR validation:

Use some modified scenarios with the added XMLs to verify goodness of HF hits

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: Geometry/HcalCommonData/data/dd4hep/cmsExtendedGeometry2021.xml
Comment (perrotta) [inline Geometry/HcalCommonData/data/dd4hep/cmsExtendedGeometry2021.xml:1]: This modifies the just merged https://github.com/cms-sw/cmssw/pull/42463: some further detail in the PR description about the fix applied can help

Comment (bsunanda) [reply Geometry/HcalCommonData/data/dd4hep/cmsExtendedGeometry2021.xml:1]: I shall use them in the updated fixed file in CMSCommonData. I am testing them and then have to redefine the scenarios

Comment (bsunanda) [reply Geometry/HcalCommonData/data/dd4hep/cmsExtendedGeometry2021.xml:1]: We have some issues in HF and we are trying to understand that. That is why we are not redefining the correct scenarios yet

**Comment by bsunanda** [comment]
I responded in the PR list. Kindly merge this. I would like to define the new scenarios now.
________________________________

File: Geometry/HcalCommonData/data/hcalsens/2021/v2/hcalsenspmf.xml
[the remark this answers is not in the document]
Comment (rappoccio) [reply Geometry/HcalCommonData/data/hcalsens/2021/v2/hcalsenspmf.xml:1]: @bsunanda this comment has not yet been addressed, can you please comment?

**Comment by bsunanda** [comment]
I need the new files in defining the modified scenario. These will balance the SD's between DDD and DD4hep.

File: Geometry/HcalCommonData/data/hcalsens/2021/v2/hcalsenspmf.xml
[the remark this answers is not in the document]
Comment (bsunanda) [reply Geometry/HcalCommonData/data/hcalsens/2021/v2/hcalsenspmf.xml:1]: The scenarios for Run3 and Phase2 will need the added files. They will be defined once this PR is merged

[the remark this answers is not in the document]
Comment (bsunanda) [reply Geometry/HcalCommonData/data/hcalsens/2021/v2/hcalsenspmf.xml:1]: The other file is used to test the dd4hep part

**Comment by rappoccio** [comment]
+1

**Pull request merged**